src/plugins/intel_npu/src/compiler_adapter/src/plugin_compiler_adapter.cpp

```diff
@@ -182,37 +176,87 @@ std::shared_ptr<IGraph> PluginCompilerAdapter::compileWS(const std::shared_ptr<o
     if (_logger.level() >= ov::log::Level::INFO) {
         compile_model_mem_start = get_peak_memory_usage();
     }
+
```

Comment:
```suggestion
storeWeightlessCacheAttribute(model);
```
One important detail I almost forgot:
Unlike the older CiP interface, the VCL one relies on a model serialization algorithm. The serialization algorithm deletes the correct values of the `WeightlessCacheAttribute`, and this attribute is important to make sure the WS flow works correctly. Therefore, just like we are doing on the CiD path, we need to store this attribute separately, and resore it in the VCL.

So, my suggestion: move [storeWeightlessCacheAttribute](https://github.com/openvinotoolkit/openvino/blob/17efa42572fcac28b14cde43c8af1ca79fc2f800/src/plugins/intel_npu/src/compiler_adapter/src/driver_compiler_adapter.cpp#L134) to a place common to both adapters (maybe a private method inside `icompiler_adapter.hpp`?) and call it at the beginning of `compileWS`, both adapters.

Note: the CI checks can pass even though this call is missing right now. I believe this is because we are not really doing "weights separation" without it, all weights end up being stored in the blob. So, the inference is accurate, but the design doesn't work as intended.

Reply:
Done! Though I've used `storeWeightlessCacheAttribute` as a public method in `icompiler_adapter.hpp`. Is there any consideration when you suggest I should use a private method? Please lmk.